refactor: Upgrade LaunchDarkly SDK and use anonymous users

jcallahan-akamai opened the pull request:
## Description 📝
- Upgrades the LD SDK to v3, which contains breaking changes (read more about [v2 -> v3 migration](https://docs.launchdarkly.com/sdk/client-side/javascript/migration-2-to-3))
- Uses anonymous users 
- - Unique keys are generated by the client SDK and stored in local storage
 
## How to test 🧪
- Run the app and ensure feature flags are evaluated appropriately

jaalah-akamai commented on packages/manager/src/IdentifyUser.tsx:
```diff
@@ -58,17 +57,15 @@ export const IdentifyUser = () => {
+            anonymous: true,
             country,
             custom: {
               taxID: _taxID,
```
Approved! This doesn't really make a difference, if we wanted to leave as is, but we can move taxID to the root now: `In version 3.0, you do not need to add custom attributes within a custom object.`

abailly-akamai commented on packages/manager/src/IdentifyUser.tsx:
```diff
@@ -58,17 +57,15 @@ export const IdentifyUser = () => {
         : account?.tax_id;
-
       if (client && userID && country && username && _taxID) {
         client
```
Unrelated to the version update, but this feels like a good candidate for a try/catch block. not high on this change but since we never used that API version before may be worth considering

jaalah-akamai replied:
@abailly-akamai Just to confirm, are you asking for something additional than the catch block we're already doing for the `client.identity`? https://github.com/linode/manager/pull/9285/files/9ae0dbfe5c8b7c5d019d80b0c77ec53ace0a5fce#diff-bce16ff1b4dbf5a6d7e4637a4277c38540996a930370da1c953d9f7793caf834R76

abailly-akamai replied:
Ah no, that's good to go!

abailly-akamai reviewed (commented):
Looks good! left a couple comments

I have however some concerns for A/B features related to resources. Imagine in the future we want to roll out a new feature, such as a new service (not just a visual change). When switching devices could encounter some serious problems with their account if they end up being in a different treatment than when opting in the feature. I don't know if this is an edge case, and we can cross that bridge later, but I think we ought to give it some thoughts

jaalah-akamai commented:
@abailly-akamai should we add a changeset to this or consider this a `chore`?

abailly-akamai commented:
@jaalah-akamai Good question. I would add one for sure because it contains breaking changes

jcallahan-akamai commented on packages/manager/src/IdentifyUser.tsx:
```diff
@@ -52,20 +52,19 @@ export const IdentifyUser = () => {
-              taxID: _taxID,
-            },
+            taxID,
+            privateAttributeNames: ['country, taxID'],
```
@abailly-akamai @jaalah-akamai I added this back just in case.

jcallahan-akamai commented:
@abailly-akamai @jaalah-akamai @cpathipa I addressed some feedback and removed the provider from the local dev tools, since it was causing a bunch of extra LaunchDarkly requests (and is seemingly not needed anymore).

jaalah-akamai commented on packages/manager/src/IdentifyUser.tsx:
```diff
@@ -53,22 +52,19 @@ export const IdentifyUser = () => {
-            },
-            privateAttributeNames: ['country', 'taxID'],
+            taxID,
+            privateAttributeNames: ['country, taxID'],
```
```suggestion
            privateAttributes: ['country, taxID'],
```

jaalah-akamai replied:
Ok we'll just have to update the name of the key - it changed with v3

bnussman-akamai commented:
Can we merge this?
